Classes/Updater.lua

```
@@ -11,129 +11,74 @@ cUpdater = {}
-		
-		-- Check if the received data size matches that that was expected
-		if (Data:len() ~= tonumber(Header["content-length"] or -1)) then
-			if (g_DownloadAttempts < g_Config.Updates.NumAttempts) then
-				g_DownloadAttempts = g_DownloadAttempts + 1
-				cUpdater:DownloadLatestVersion()
+	cUrlClient:Get("https://raw.githubusercontent.com/cuberite/WorldEdit/master/Info.lua",
+		function(a_Body, a_Data)
+			if (a_Body) then
+				cUpdater:ParsePluginInfo(a_Body)
 			else
-				LOGWARNING("Error while downloading newer worldedit version")
+				-- The request failed, schedule a retry
+				cRoot:Get():GetDefaultWorld():QueueTask(
+					function()
+						cUpdater:CheckForNewerVersion()
+					end
+				)
 			end
-			return
 		end
-		
-		-- Write the ZIP data to the file. The filename looks like this: "WorldEdit v<DisplayVersion>.zip"
-		local ZipFile = assert(io.open("Plugins/WorldEdit v" .. (g_LatestVersion or "_Unknown") .. ".zip", "wb"), "Failed to open \"Plugins/WorldEdit v" .. (g_LatestVersion or "_Unknown") .. ".zip\"")
-		ZipFile:write(Data)
-		ZipFile:close()
-		
-		LOGINFO(string.format("New WorldEdit version downloaded to %q", "Plugins/WorldEdit v" .. (g_LatestVersion or "_Unknown") .. ".zip"))
-	end,	
-}
-
-
-
-
-
-function cUpdater:CheckForNewerVersion()
-	cNetwork:Connect("raw.githubusercontent.com", 443, VersionCheckForNewerVersionerCallbacks)
+	)
 end
 
 
 
 
 
-function cUpdater:DownloadLatestVersion()
-	cNetwork:Connect("codeload.github.com", 443, NewVersionDownloaderCallbacks)
+function cUpdater:DownloadLatestVersion(a_DisplayVersion)
+	if (cUpdater.s_DownloadAttempts > g_Config.Updates.NumAttempts) then
+		LOGWARNING("Error while downloading newer worldedit version")
+		return
+	end
+	cUpdater.s_DownloadAttempts = cUpdater.s_DownloadAttempts + 1;
+	
+	cUrlClient:Get("https://raw.githubusercontent.com/cuberite/WorldEdit/zip/master", 
+		function (a_Body, a_Data)
+			if (not a_Body or (a_Body:len() ~= tonumber(a_Data["Content-Length"]))) then
```

It should work - the content-length is used (if the response isn't chunked) to distinguish the response's end and the potential beginning of another response over the same connection (we don't use that yet, though), so if the server cuts the answer short, the parser should report an error.